version: "3.9"

volumes:
  ipython_history: {}

services:
  postgres:
    image: postgres:12-alpine
    ports:
      - "5432:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -h postgres -t 5 -U ${COMPOSE_PROJECT_NAME}-user || false"]
      interval: 10s
      timeout: 5s
      retries: 5
    environment:
      - POSTGRES_DB=${COMPOSE_PROJECT_NAME}-dev
      - POSTGRES_USER=${COMPOSE_PROJECT_NAME}-user
      - POSTGRES_PASSWORD=manager

  web:
    build:
      context: .
      args:
        install_groups: "main,dev,test,local"
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - .:/workspace
    ports:
      - '8000:8000'
    env_file:
      - .env
    depends_on:
      postgres:
          condition: service_healthy